# Chi-Square test

In [71]:
import pandas as pd
import numpy as np

In [73]:
Satisfaction = ["Verysatisfied", "Satisfied", "Neutral", "Unsatisfied", "VeryUnsatisfied","Total"]
SmartThermostat = [50,80,60,30,20,240]
SmartLight = [70,100,90,50,50,360]
Total=[120,180,150,80,70,600]

In [75]:
data=pd.DataFrame({'Satisfaction':Satisfaction, 'Smart Thermostat': SmartThermostat, 'SmartLight': SmartLight, 'Total':Total})
data

Satisfaction  Smart Thermostat  SmartLight  Total
0    Verysatisfied                50          70    120
1        Satisfied                80         100    180
2          Neutral                60          90    150
3      Unsatisfied                30          50     80
4  VeryUnsatisfied                20          50     70
5            Total               240         360    600

# 1.State the Hypothesis:

### Null Hypothesis (H₀): There is no association between device type and customer satisfaction (independent).

### Alternative Hypothesis (H₁): There is an association between device type and customer satisfaction (dependent).

#  2.Compute the Chi-Square Statistic

In [77]:
observed = np.array([
    [50 ,70],
    [80, 100],
    [60, 90],
    [30,50],
    [20,50]
])

In [79]:
import scipy.stats as  stats
stats.chi2_contingency(observed)

Chi2ContingencyResult(statistic=5.638227513227513, pvalue=0.22784371130697179, dof=4, expected_freq=array([[ 48.,  72.],
       [ 72., 108.],
       [ 60.,  90.],
       [ 32.,  48.],
       [ 28.,  42.]]))

In [81]:
from scipy.stats import chi2_contingency
chi, p, dof, expected = chi2_contingency(observed)

In [84]:
print(chi,p,dof)

5.638227513227513 0.22784371130697179 4


# Determine the Critical Value

In [1]:
import scipy.stats as stats

alpha = 0.05
df = 4

critical_value = stats.chi2.ppf(1 - alpha, df)

print("Critical Value:", critical_value)


Critical Value: 9.487729036781154


# 4.Make a Decision

## Compare p-value with Alpha (0.05)
## p-value = 0.2278
## Alpha (significance level) = 0.05
## Since p-value (0.2278) > 0.05, we fail to reject H₀.
## We accepted the Null Hypothesis